fri/src/fold_even_odd.rs

```diff
@@ -26,55 +28,27 @@ pub fn fold_even_odd<F: TwoAdicField>(poly: &[F], beta: F) -> Vec<F> {
     //     result(g^(2i)) = p_e(g^(2i)) + beta p_o(g^(2i))
     //                    = (1/2 + beta/2 g_inv^i) p(g^i)
     //                    + (1/2 - beta/2 g_inv^i) p(g^(n/2 + i))
-
-    let n = poly.len();
-    debug_assert!(n > 1);
-
-    let log_n = log2_strict_usize(n);
-
-    let g_inv = F::two_adic_generator(log_n).inverse();
+    let m = RowMajorMatrix::new(poly, 2);
+    let g_inv = F::two_adic_generator(log2_strict_usize(m.height()) + 1).inverse();
     let one_half = F::two().inverse();
     let half_beta = beta * one_half;
 
     // beta/2 times successive powers of g_inv
-    let powers = g_inv.shifted_powers_packed::<F::Packing>(half_beta);
```

Comment:
Looks like vectorization was removed for now? Sounds fine if so, getting it working is more important, maybe just add a TODO to re-vectorize later